File: leetcode/38.js
// 38. Count and Say

/**
 * @param {number} n
 * @return {string}
 */
var countAndSay = function(n) {
    if (n === 1) return "1";
    let ans = ""
    let prevAns = countAndSay(n-1);
    let prevNum = prevAns[0];
    let count = 1;
    for (let i = 1; i < prevAns.length; i++) {
        let curNum = prevAns[i]
        if ( curNum === prevNum) {
            count++;
        } else {
            ans = ans.concat(count, prevNum);
            prevNum = curNum;
            count = 1;
        }
    }
    ans = ans.concat(count, prevNum);
    return ans
};

File: leetcode/38.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return "1"
        ans = ""
        prev_ans = self.countAndSay(n - 1)
        prev_num = prev_ans[0]
        count = 1
        for i, n in enumerate(prev_ans):
            if i == 0:
                continue
            if prev_num == n:
                count += 1
            else:
                ans =  ans + f"{count}{prev_num}"
                count = 1
                prev_num = n
        ans = ans + f"{count}{prev_num}"
        return ans